packages/yourloops/components/device/use-device.hook.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023, Diabeloop
+ *
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {
+  type CgmConfig,
+  type DeviceConfig,
+  type ParameterConfig,
+  type ParametersChange,
+  type PumpConfig,
+  type PumpSettings
+} from 'medical-domain'
+import type MedicalDataService from 'medical-domain'
+import { useTranslation } from 'react-i18next'
+import textTable from 'text-table'
+import { formatParameterValue } from './utils/device.utils'
+import { useEffect } from 'react'
+import moment from 'moment'
+
+interface UseDeviceSettingsReturn {
+  cgm: CgmConfig
+  copySettingsToClipboard: () => Promise<void>
+  device: DeviceConfig
+  history: ParametersChange[]
+  lastUploadDate: string
+  parameters: ParameterConfig[]
+  pump: PumpConfig
+}
+
+export const useDevice = (medicalData: MedicalDataService): UseDeviceSettingsReturn => {
```

Comment:
This could all be defined in a util file. There is no state. The only hook used is `t` which is accessible without react hook system. The useeffect would be in the component calling this. There wouldn't be side effects that way (the way the variables `parameters` is handled instead of being returned by the function that use it).
IMO this would remove complexity and this react layer that we do not need :thinking: 

<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏Any opinion on this?